packages/widgets/src/_inputs/widget-select-input.tsx

```diff
@@ -25,14 +29,22 @@ export const WidgetSelectInput = ({
   kind,
   options,
 }: CommonWidgetInputProps<"select">) => {
-  const t = useWidgetInputTranslation(kind, property);
+  const t = useI18n();
+  const tWidget = useWidgetInputTranslation(kind, property);
   const form = useFormContext();
 
   return (
     <Select
-      label={t("label")}
-      data={options.options as unknown as SelectOption[]}
-      description={options.withDescription ? t("description") : undefined}
+      label={tWidget("label")}
+      data={options.options.map((option) =>
+        typeof option === "string"
+          ? option
+          : {
+              value: option.value,
+              label: translateIfNecessary(t, option.label)!,
+            },
+      )}
+      description={options.withDescription ? tWidget("description") : undefined}
```

Comment:
If the tWidget(description) is correctly typed you can do description={tWidget("description") ?? undefined)

Reply:
No because it would use the key as fallback and not undefined or null